tests/performance/performance.cpp

```diff
@@ -90,21 +136,105 @@ typedef class _ebpf_hash_table_test_state
         }
     }
 
+    void
+    test_replace_value_overlap()
+    {
+        uint64_t value = 12345678;
+        // Update conflicting keys
+        for (auto& key : keys) {
+            ebpf_epoch_enter();
+            ebpf_hash_table_update(
+                table,
+                reinterpret_cast<uint8_t*>(&key),
+                reinterpret_cast<uint8_t*>(&value),
+                nullptr,
+                EBPF_HASH_TABLE_OPERATION_REPLACE);
+            ebpf_epoch_exit();
+        }
+    }
+
     size_t
     multiplier()
     {
         return keys.size();
     }
 
-    ~_ebpf_hash_table_test_state() { ebpf_hash_table_destroy(table); }
-
   private:
     ebpf_hash_table_t* table;
     std::vector<uint32_t> keys;
+    bool platform_initiated = false;
+    bool epoch_initated = false;
+    uint32_t cpu_count;
 
 } ebpf_hash_table_test_state_t;
 
+typedef class _ebpf_program_test_state
+{
+  public:
+    _ebpf_program_test_state(std::vector<ebpf_instruction_t> byte_code)
+        : byte_code(byte_code), program_info_provider(EBPF_PROGRAM_TYPE_XDP)
+    {
+        ebpf_program_parameters_t parameters = {EBPF_PROGRAM_TYPE_XDP};
+        REQUIRE(ebpf_core_initiate() == EBPF_SUCCESS);
+        REQUIRE(ebpf_program_create(&program) == EBPF_SUCCESS);
+
+        REQUIRE(ebpf_program_initialize(program, &parameters) == EBPF_SUCCESS);
+    }
+    ~_ebpf_program_test_state()
+    {
+        ebpf_object_release_reference(reinterpret_cast<ebpf_object_t*>(program));
+        ebpf_core_terminate();
+    }
+
+    void
+    setup_jit_program()
```

Comment:
```suggestion
    set_up_jit_program()
```
"set up" is a verb.  "setup" is a noun.